### ISSUE TITLE
Question about LC0 sharpness calcuation

### ISSUE THREAD

**Issue by Charles-Wu-Chen** [jk182/chessProjects#1]
https://github.com/jk182/chessProjects/blob/7bbb6f4944947fc0f43146b16ff6031cd2cb18ad/functions.py#L76

https://x.com/LeelaChessZero/status/1637763896596463616

this line of code is different with the one suggested  by LC0 .  I am not math expert,  but wondering why there is max( ) to cut off all the negative values (which can be a very big negative double). 

Thanks in advance for the answers

**Comment by jk182** [comment]
Yes, this is meant to avoid negative values which would then be counted as sharp positions due to the square. In general, this formula doesn't work well for cases where one side has a big advantage, so I tried different things to avoid this but found nothing elegant yet.
